### ISSUE TITLE
RazorGenerator.MVC Update MVC version

### ISSUE THREAD

**Issue by vallieresc** [RazorGenerator/RazorGenerator#160]
I'm working on a .Net project running on latest MVC version (5.2.3.0) and when executing assembly.GetTypes() I get a ReflectionTypeLoadException when loading System.Web.Mvc 3.0.0.1 from RazorGenerator.MVC dll. This is when executing my T4 template. 

Any chance MVC can get updated to latest like your MsBuild project?

**Comment by pranavkm** [comment]
Does your web.config have a binding redirect to the 5.2.3 version of Mvc? Typically that should take care of the version conflict.

**Comment by vallieresc** [comment]
`<dependentAssembly>
        <assemblyIdentity name="System.Web.Mvc" publicKeyToken="31bf3856ad364e35" />
        <bindingRedirect oldVersion="0.0.0.0-5.2.3.0" newVersion="5.2.3.0" />
      </dependentAssembly>`

**Comment by vallieresc** [comment]
I don't believe the web.config binding redirect is read during T4 template execution.

**Comment by pranavkm** [comment]
That seems problematic. We'd have to make a version of RazorGenerator.MsBuild compiled against each available release of Mvc. https://stackoverflow.com/questions/45628571/c-sharp-t4-using-assemblybinding-bindingredirect-from-web-config suggests putting the redirects in the devenv.exe.config which admittedly is super gross. 

What do other libraries do to that build on top of Mvc to support this?

**Comment by vallieresc** [comment]
The project I am working on might be supported by multiple developers so a machine specific setting is a bad idea. I haven't had any issues with other libraries.

**Comment by vallieresc** [comment]
Isn't this possible: https://shazwazza.com/post/multi-targeting-a-single-net-project-to-build-for-different-framework-versions/

**Comment by pranavkm** [comment]
Seems like they're pivoting on TFM to determine the dependency which isn't entirely correct. Mvc 3 should work on .NET4.5 just fine and we wouldn't want to update consumers of the package on account of our reference.
